templates_test.go

```diff
@@ -114,45 +140,58 @@ func TestWriteDefault(t *testing.T) {
 	}
 }
 
-// TestWriteInvalid ensures that specifying unrecgognized runtime/template errors
-func TestWriteInvalid(t *testing.T) {
+// TestTemplateInvalidErrors ensures that specifying unrecgognized
+// runtime/template errors
+func TestTemplateInvalidErrors(t *testing.T) {
 	// create test directory
 	root := "testdata/testWriteInvalid"
 	defer using(t, root)()
 
-	w := templateWriter{}
-	var err error // should be populated with the correct error type
+	client := fn.New(fn.WithRegistry(TestRegistry))
+
+	// Error will be type-checked.
+	var err error
 
 	// Test for error writing an invalid runtime
-	// (the http template
-	err = w.Write("invalid", DefaultTemplate, root)
-	if !errors.Is(err, ErrRuntimeNotFound) {
+	err = client.Create(fn.Function{
+		Root:    root,
+		Runtime: "invalid",
+	})
+	if !errors.Is(err, fn.ErrRuntimeNotFound) {
 		t.Fatalf("Expected ErrRuntimeNotFound, got %v", err)
 	}
 
 	// Test for error writing an invalid template
-	err = w.Write(TestRuntime, "invalid", root)
-	if !errors.Is(err, ErrTemplateNotFound) {
+	err = client.Create(fn.Function{
+		Root:     root,
+		Runtime:  TestRuntime,
+		Template: "invalid",
+	})
+	if !errors.Is(err, fn.ErrTemplateNotFound) {
 		t.Fatalf("Expected ErrTemplateNotFound, got %v", err)
 	}
 }
 
-// TestWriteModeEmbedded ensures that templates written from the embedded
+// TestTemplateModeEmbedded ensures that templates written from the embedded
 // templates retain their mode.
-func TestWriteModeEmbedded(t *testing.T) {
+func TestTemplateModeEmbedded(t *testing.T) {
 	if runtime.GOOS == "windows" {
 		return
 		// not applicable
 	}
 
 	// set up test directory
-	var err error
 	root := "testdata/testWriteModeEmbedded"
```

Comment:
I bet you see where this is going :)
```suggestion
	root := "testdata/testTemplateModeEmbedded"
```